todoCLI/todoCLI.js

```diff
@@ -0,0 +1,494 @@
+//To Do CLI function
+const toDoCli = (ClassTodo, openFileName, newFileName) => {
+    let newList = new ClassTodo(newFileName)
+    if (openFileName) {
+        openFileName.forEach(item => newList.items.push(item))
+    }
+    console.log("Welcome to Todo CLI!\n\n--------------------\n")
+    newList.menu()
+    terminal.on('line', function (input) {
+        if (input === 'v') {
+            //Pressing v then Enter will display the contents of the todo list
+            newList.view()
+        } else if (input === 'n') {
+            //Pressing n then Enter will ask the user what item to add to the list
+            terminal.question("What?\n", function (item) {
+                //User's input will be added to the list
+                newList.add(item)
+            })
+        } else if (/^c[0-9]+$/.test(input)) {
+            //Pressing cX where X refers to the index of a Todo item then Enter 
+            //should mark that item as complete.
+            newList.complete(input.substring(1))
+        } else if (/^d[0-9]+$/.test(input)) {
+            //Pressing dX where X refers to the index of a Todo item then `Enter` should remove that from the list. 
+            newList.deleted(input.substring(1))
+        } else if (input === 'q') {
+            //Pressing q quits the application
+            newList.quit()
+            process.exit(1)
+        } else if (input === 's') {
+            //Stretch: Save to File
+            if (newList.newFileName) {
+                // If a filePath argument has been given, use it as a suggested filePath to save the file.
+                terminal.question(`Where? (${newList.newFileName})\n`, function (item) {
+                    newList.save(item)
+                })
+            } else {
+                terminal.question("Where?\n", function (item) {
+                    newList.save(item)
+                })
+            }
+        } else {
+            newList.menu()
+        }
+    })
+}
+
+// -------------------------------------------------------------------------------------------------------------------------------------
+// Stretch: Open File
+//Read Directory function that returns a promise
+function readdir(pathName) {
+    return new Promise((resolve, reject) => {
+        fs.readdir(pathName, (err, files) => {
+            if (err)
+                reject(err);
+            else {
+                resolve(files)
+            }
+        })
+    })
+}
+//Read File function that returns a promise
+function readFile(fileName) {
+    return new Promise((resolve, reject) => {
+        fs.readFile(fileName, { encoding: 'utf8' }, (err, data) => {
+            if (err) {
+                reject(err)
+            } else {
+                resolve(data)
+            }
+        })
+    })
+}
+//Write File function that returns a promise
+function writeFile(fileName, content) {
+    return new Promise((resolve, reject) => {
```

Comment:
Was your intention to use `fs/promises`? Either way, using promise here is not necessary with fs callback.